amo/tx/stake.go

```diff
@@ -53,25 +52,24 @@ func (t *TxStake) Execute(store *store.Store) (uint32, string, []tm.KVPair) {
 		return code.TxCodeBadParam, err.Error(), nil
 	}
 
+	if !txParam.Amount.GreaterThan(zero) {
+		return code.TxCodeUnavailableAmount, "unavailable amount", nil
+	}
+
 	balance := store.GetBalance(t.GetSender(), false)
 	if balance.LessThan(&txParam.Amount) {
 		return code.TxCodeNotEnoughBalance, "not enough balance", nil
 	}
 
 	balance.Sub(&txParam.Amount)
-	stake := store.GetStake(t.GetSender(), false)
-	if stake == nil {
-		var k ed25519.PubKeyEd25519
-		copy(k[:], txParam.Validator)
-		stake = &types.Stake{
-			Amount:    txParam.Amount,
-			Validator: k,
-		}
-	} else if bytes.Equal(stake.Validator[:], txParam.Validator[:]) {
```

Comment:
we should not add something, but we should check validator match and return BadValidator or ValidatorMismatch when it does not match. need to check somewhere. check these:
- stake owner mismatch (permission denied)
- validator key mismatch (validator mismatch)